diff --git a/src/better_sqlite3.cpp b/src/better_sqlite3.cpp
index 13f054a641f63dda3ebbf71ea5613cf12b054fad..2c323c65c76ff1c231eda79101a4632068b2ed9c 100644
--- a/src/better_sqlite3.cpp
+++ b/src/better_sqlite3.cpp
@@ -57,7 +57,7 @@ NODE_MODULE_INIT(/* exports, context */) {
 
 	// Initialize addon instance.
 	Addon* addon = new Addon(isolate);
-	v8::Local<v8::External> data = v8::External::New(isolate, addon);
+	v8::Local<v8::External> data = EXTERNAL_NEW(isolate, addon);
 	node::AddEnvironmentCleanupHook(isolate, Addon::Cleanup, addon);
 
 	// Create and export native-backed classes and functions.
diff --git a/src/util/helpers.cpp b/src/util/helpers.cpp
index f7771520075e2dc871d1d60b801bbbb540c6c23d..db7fd9f563b3984f88b3c452e30b8c9eca2bf106 100644
--- a/src/util/helpers.cpp
+++ b/src/util/helpers.cpp
@@ -89,7 +89,7 @@ void SetPrototypeGetter(
 	recv->InstanceTemplate()->SetNativeDataProperty(
 		InternalizedFromLatin1(isolate, name),
 		func,
-		0,
+		nullptr,
 		data
 	);
 }
diff --git a/src/util/macros.cpp b/src/util/macros.cpp
index e75d53763e575cb1ec5f560d1303100ebb380249..2bd490a1656e566afa36d0606f26af5fa53cff5c 100644
--- a/src/util/macros.cpp
+++ b/src/util/macros.cpp
@@ -27,7 +27,14 @@
 #define EasyIsolate v8::Isolate* isolate = v8::Isolate::GetCurrent()
 #define OnlyIsolate info.GetIsolate()
 #define OnlyContext isolate->GetCurrentContext()
-#define OnlyAddon static_cast<Addon*>(info.Data().As<v8::External>()->Value())
+#if defined(NODE_MODULE_VERSION) && NODE_MODULE_VERSION >= 146
+#define EXTERNAL_NEW(isolate, value) v8::External::New((isolate), (value), 0)
+#define EXTERNAL_VALUE(value) (value)->Value(0)
+#else
+#define EXTERNAL_NEW(isolate, value) v8::External::New((isolate), (value))
+#define EXTERNAL_VALUE(value) (value)->Value()
+#endif
+#define OnlyAddon static_cast<Addon*>(EXTERNAL_VALUE(info.Data().As<v8::External>()))
 #define UseIsolate v8::Isolate* isolate = OnlyIsolate
 #define UseContext v8::Local<v8::Context> ctx = OnlyContext
 #define UseAddon Addon* addon = OnlyAddon
