diff --git a/node_modules/@sentry/browser/dist/sdk.js b/node_modules/@sentry/browser/dist/sdk.js
index bf01e8c..aadd945 100644
--- a/node_modules/@sentry/browser/dist/sdk.js
+++ b/node_modules/@sentry/browser/dist/sdk.js
@@ -12,7 +12,6 @@ exports.defaultIntegrations = [
     new integrations_1.Breadcrumbs(),
     new integrations_1.GlobalHandlers(),
     new integrations_1.LinkedErrors(),
-    new integrations_1.Dedupe(),
     new integrations_1.UserAgent(),
 ];
 /**
