diff --git a/node_modules/eslint-config-react-app/index.js b/node_modules/eslint-config-react-app/index.js
index f83ed06..b0857ce 100644
--- a/node_modules/eslint-config-react-app/index.js
+++ b/node_modules/eslint-config-react-app/index.js
@@ -132,13 +132,6 @@ module.exports = {
       },
     ],
     'no-unused-labels': 'warn',
-    'no-unused-vars': [
-      'warn',
-      {
-        args: 'none',
-        ignoreRestSiblings: true,
-      },
-    ],
     'no-use-before-define': [
       'warn',
       {
