diff --git a/node_modules/pikaday/pikaday.js b/node_modules/pikaday/pikaday.js
index 8dae838..13ecfbe 100755
--- a/node_modules/pikaday/pikaday.js
+++ b/node_modules/pikaday/pikaday.js
@@ -12,8 +12,8 @@
     if (typeof exports === 'object') {
         // CommonJS module
         // Load moment.js as an optional dependency
-        try { moment = require('moment'); } catch (e) {}
-        module.exports = factory(moment);
+        // try { moment = require('moment'); } catch (e) {}
+        module.exports = factory(null);
     } else if (typeof define === 'function' && define.amd) {
         // AMD. Register as an anonymous module.
         define(function (req)
