diff --git a/node_modules/react-dnd-html5-backend/dist/esm/HTML5Backend.js b/node_modules/react-dnd-html5-backend/dist/esm/HTML5Backend.js
index 942091b..7378aa0 100644
--- a/node_modules/react-dnd-html5-backend/dist/esm/HTML5Backend.js
+++ b/node_modules/react-dnd-html5-backend/dist/esm/HTML5Backend.js
@@ -280,9 +280,9 @@ function () {
         // Show user-specified drop effect.
         e.preventDefault();
 
-        if (e.dataTransfer) {
-          e.dataTransfer.dropEffect = _this.getCurrentDropEffect();
-        }
+        // if (e.dataTransfer) {
+        //   e.dataTransfer.dropEffect = _this.getCurrentDropEffect();
+        // }
       } else if (_this.isDraggingNativeItem()) {
         // Don't show a nice cursor but still prevent default
         // "drop and blow away the whole document" action.
@@ -290,9 +290,9 @@ function () {
       } else {
         e.preventDefault();
 
-        if (e.dataTransfer) {
-          e.dataTransfer.dropEffect = 'none';
-        }
+        // if (e.dataTransfer) {
+        //   e.dataTransfer.dropEffect = 'none';
+        // }
       }
     };
 
