diff --git a/node_modules/nodejs-mobile-react-native/android/build.gradle b/node_modules/nodejs-mobile-react-native/android/build.gradle
index e54f25c..8b0138b 100644
--- a/node_modules/nodejs-mobile-react-native/android/build.gradle
+++ b/node_modules/nodejs-mobile-react-native/android/build.gradle
@@ -76,9 +76,6 @@ android {
     }
 
     sourceSets {
-        main {
-            jniLibs.srcDirs 'libnode/bin/'
-        }
         main.assets.srcDirs += '../install/resources/nodejs-modules'
     }
 
diff --git a/node_modules/nodejs-mobile-react-native/index.js b/node_modules/nodejs-mobile-react-native/index.js
index 92a53ae..e0ac119 100644
--- a/node_modules/nodejs-mobile-react-native/index.js
+++ b/node_modules/nodejs-mobile-react-native/index.js
@@ -1,6 +1,6 @@
 
 import { NativeModules, NativeAppEventEmitter } from 'react-native';
-var EventEmitter = require('react-native/Libraries/vendor/emitter/EventEmitter').default;
+import EventEmitter from 'react-native/Libraries/vendor/emitter/EventEmitter';
 
 const EVENT_CHANNEL = '_EVENTS_';
 
