diff --git a/node_modules/react-native-safe-area-view/index.js b/node_modules/react-native-safe-area-view/index.js
index a21ff41..4a92960 100644
--- a/node_modules/react-native-safe-area-view/index.js
+++ b/node_modules/react-native-safe-area-view/index.js
@@ -5,6 +5,7 @@ import {
   Platform,
   StyleSheet,
   Animated,
+  NativeModules
 } from 'react-native';
 import hoistStatics from 'hoist-non-react-statics';
 
@@ -86,11 +87,7 @@ const statusBarHeight = isLandscape => {
    * we do.
    */
   if (Platform.OS === 'android') {
-    if (global.Expo) {
-      return global.Expo.Constants.statusBarHeight;
-    } else {
-      return 0;
-    }
+    return NativeModules.StatusBarHeight.statusBarHeight;
   }
 
   if (isIPhoneX) {
