--- RCTInputAccessoryViewContent.m	2018-08-14 13:49:55.000000000 -0400
+++ node_modules/react-native/Libraries/Text/TextInput/RCTInputAccessoryViewContent.m	2018-08-14 13:49:56.000000000 -0400
@@ -25,19 +25,6 @@
 
 - (void)didMoveToSuperview
 {
-
-#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 110000 /* __IPHONE_11_0 */
-  // Avoid the home pill (in portrait mode)
-  // TODO: Support rotation, anchor to left and right without breaking frame x coordinate (T27974328).
-  if (@available(iOS 11.0, *)) {
-    if (self.window) {
-      [_safeAreaContainer.bottomAnchor
-       constraintLessThanOrEqualToSystemSpacingBelowAnchor:self.window.safeAreaLayoutGuide.bottomAnchor
-       multiplier:1.0f].active = YES;
-    }
-  }
-#endif
-
 }
 
 - (void)setFrame:(CGRect)frame
