// @docusaurus/theme-common/lib/index.js
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
/*
 * APIs to document
 */
export { useThemeConfig, } from './utils/useThemeConfig';
export { createStorageSlot, listStorageKeys } from './utils/storageUtils';
export { useContextualSearchFilters } from './utils/searchUtils';
export { useCurrentSidebarCategory, filterDocCardListItems, } from './utils/docsUtils';
export { usePluralForm } from './utils/usePluralForm';
export { useCollapsible, Collapsible } from './components/Collapsible';
export { ThemeClassNames } from './utils/ThemeClassNames';
export { useIsomorphicLayoutEffect, useEvent, usePrevious, composeProviders, ReactContextError, } from './utils/reactUtils';
export { PageMetadata, HtmlClassNameProvider } from './utils/metadataUtils';
export { useColorMode } from './contexts/colorMode';
export { NavbarSecondaryMenuFiller, } from './contexts/navbarSecondaryMenu/content';
export { useWindowSize } from './hooks/useWindowSize';
/*
 * APIs kept undocumented, on purpose
 * Note: we still guarantee retro-compatibility on those
 */
export { translateTagsPageTitle, listTagsByLetters, } from './utils/tagsUtils';
export { isMultiColumnFooterLinks } from './utils/footerUtils';
export { isRegexpStringMatch } from './utils/regexpUtils';
export { duplicates, uniq } from './utils/jsUtils';
export { usePrismTheme } from './hooks/usePrismTheme';
export { useDocsPreferredVersion } from './contexts/docsPreferredVersion';
