diff --git a/node_modules/@rocket.chat/peggy-loader/dist/cjs/index.js b/node_modules/@rocket.chat/peggy-loader/dist/cjs/index.js
index 8b5ebab..33234ab 100644
--- a/node_modules/@rocket.chat/peggy-loader/dist/cjs/index.js
+++ b/node_modules/@rocket.chat/peggy-loader/dist/cjs/index.js
@@ -16,8 +16,7 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 Object.defineProperty(exports, "__esModule", { value: true });
 var peggy_1 = __importDefault(require("peggy"));
 function peggyLoader(grammarContent) {
-    var options = __assign({ format: 'commonjs' }, this.getOptions());
+    var options = { format: 'commonjs' };
     return peggy_1.default.generate(grammarContent, __assign({ output: 'source' }, options));
 }
 exports.default = peggyLoader;
-//# sourceMappingURL=index.js.map
\ No newline at end of file
